src/libutil/util.cc

```diff
@@ -54,6 +54,16 @@ std::optional<std::string> getEnv(const std::string & key)
     return std::string(value);
 }
 
+std::optional<std::string> getEnvNonEmpty(const std::string & key) {
+    auto value = getEnv(key);
+    if (value == "") {
+        // TODO: determine whether this should be a warning or an error.
+        warn("ignoring the '%1%' env variable, its value has been set to \"\"", key);
```

Comment:
I would drop this warning. If empty vars are equivalent to the var not being set, there is no reason to complain about it.

Reply:
They are not equal on my system at least:

```
philip@shiki ~> env FOO="" printenv FOO

philip@shiki ~> env printenv FOO
philip@shiki ~ [1]>
```

one returns `""` and the other status 1

Reply:
If I had something set to `""`, I’d be kind of confused if nix ignored it silently.